test/controllers/declaration/AdditionalDocumentsRequiredControllerSpec.scala

```diff
@@ -0,0 +1,139 @@
+import org.mockito.Mockito.{reset, verify, when}
+import org.scalatest.Assertion
+import play.api.data.Form
+import play.api.libs.json.Json
+import play.api.mvc.{AnyContentAsEmpty, Call, Request}
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import views.html.declaration.additionalDocuments.additional_documents_required
+
+class AdditionalDocumentsRequiredControllerSpec extends ControllerSpec {
+
+  private val page = mock[additional_documents_required]
+
+  val controller = new AdditionalDocumentsRequiredController(
+    mockAuthAction,
+    mockJourneyAction,
+    mockExportsCacheService,
+    navigator,
+    stubMessagesControllerComponents(),
+    page
+  )(ec)
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    authorizedUser()
+    when(page.apply(any[Mode], any[String], any[Form[YesNoAnswer]])(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(page)
+    super.afterEach()
+  }
+
+  val itemId = "itemId"
+
+  override def getFormForDisplayRequest(request: Request[AnyContentAsEmpty.type]): Form[_] = {
+    withNewCaching(aDeclaration(withItem(anItem(withItemId(itemId)))))
+    await(controller.displayPage(Mode.Normal, itemId)(request))
+    theResponseForm
+  }
+
+  def theResponseForm: Form[YesNoAnswer] = {
+    val captor = ArgumentCaptor.forClass(classOf[Form[YesNoAnswer]])
+    verify(page).apply(any[Mode], any[String], captor.capture())(any(), any())
+    captor.getValue
+  }
+
+  "AdditionalDocumentsRequiredController" should {
+
+    onEveryDeclarationJourney() { implicit request =>
+      "return 200 (OK)" when {
+
+        "display page method is invoked and cache is empty" in {
+          withNewCaching(aDeclarationAfter(request.cacheModel, withItem(anItem(withItemId(itemId)))))
+          val result = controller.displayPage(Mode.Normal, itemId)(getRequest())
+          status(result) must be(OK)
+          verifyPageInvoked
+        }
+
+        "display page method is invoked and cache is not empty" in {
+          val documentsProducedData = DocumentsProducedData(Some(YesNoAnswer.Yes), Seq.empty)
+          withNewCaching(
+            aDeclarationAfter(request.cacheModel, withItem(anItem(withItemId(itemId), withDocumentsProducedData(documentsProducedData))))
+          )
+
+          val result = controller.displayPage(Mode.Normal, itemId)(getRequest())
+          status(result) must be(OK)
+          verifyPageInvoked
+        }
+      }
+
+      "return 303 (SEE_OTHER) and redirect to the 'Additional Documents' page" when {
+        "answer is 'yes'" in {
+          verifyRedirect(Some(YesNoAnswers.yes), routes.AdditionalDocumentsAddController.displayPage(Mode.Normal, itemId))
+        }
+      }
+
+      "return 303 (SEE_OTHER) and redirect to the 'Items Summary' page" when {
+        "answer is 'no'" in {
+          verifyRedirect(Some(YesNoAnswers.no), routes.ItemsSummaryController.displayItemsSummaryPage())
```

Comment:
Using this method makes it very unclear about what this test is actually testing. Especially, which method is it testing? Because in this method both controller endpoints are used.